_pytest/doctest.py

```diff
@@ -103,8 +105,21 @@ def setup(self):
 
     def runtest(self):
         _check_all_skipped(self.dtest)
+        self._disable_output_capturing_for_darwin()
         self.runner.run(self.dtest)
 
+    def _disable_output_capturing_for_darwin(self):
+        """
+        Disable output capturing. Otherwise, stdout is lost to doctest (#985)
+        """
+        if platform.system() != 'Darwin':
+            return
+        capman = self.config.pluginmanager.getplugin("capturemanager")
+        if capman:
+            out, err = capman.suspend_global_capture(in_=True)
+            sys.stdout.write(out)
+        sys.stdout.write(err)
```

Comment:
err might be undefinded here if bool(capman) is False.

Reply:
Hmm. I'd consider that contingency unlikely (negligible) unless capture managers are expected to present as `__bool__() == False` under some condition.

Reply:
if the capturemanager is disabled by a cli option the code would fail
please lets use consisted indentation for the related code

for overly pedantic correctness it might make sense to change the check to `is None`

Reply:
I agree, no harm being consistent. Btw, shouldn't the last line write to `sys.stderr` instead?

Reply:
> if the capturemanager is disabled by a cli option the code would fail

I'm afraid I'm not following. Do you mean `--capture=no`? As I understand it, passing that parameter would mean that `bool(capman) == True`. And I tested it, and it [works fine](https://gist.github.com/5752f7bef1b8fe80cbd0cbb339c2970d).

I think I see what you're saying - that on line 121, `err` won't have been defined if the capman was False. And that would happen by disabling the plugin entirely, such as with `-p no:capturemanager`. Testing that, I get [this output](https://gist.github.com/bc5379aefa337c95376fa8b3139c2cf8). It's not quite what I expected. The code didn't fail anywhere near doctest.py.

> please lets use consisted indentation for the related code

At first, I thought this was a stylistic comment, but I see now that you've recognized an error I introduced when I copied this code from the original.

> for overly pedantic correctness it might make sense to change the check to is None

I'm not sure about that. It may be the case that if the capture manager defines `__bool__` which returns False, that's also a condition that should be bypassed (correctly). Since it doesn't define it either way, the boolean test currently seems to me to be correct either way, at which point I prefer the shorter code.

> Btw, shouldn't the last line write to sys.stderr instead?

Yes, that does seem incongruent. I'll test with this suggestion.

Reply:
> err might be undefinded here if bool(capman) is False.

Now that I re-read the original comment, I see that was the sole concern, and I was on an unrelated tangent.